types/src/v0/impls/state.rs

```diff
@@ -702,18 +698,29 @@ impl HotShotState<SeqTypes> for ValidatedState {
                 proposed_finalized.number()
             );
         }
-        // Wait for l1_finalized with a timeout.
         let finalized = instance
             .l1_client
             .wait_for_finalized_block(proposed_finalized.number())
             .await;
 
-        if finalized.hash() != proposed_finalized.hash() {
-            tracing::error!("Invalid proposal: l1_finalized hash mismatch");
+        if finalized != proposed_finalized {
+            tracing::error!("Invalid proposal: l1_finalized mismatch");
             return Err(BlockError::InvalidBlockHeader);
         }
-        if finalized.timestamp() != proposed_finalized.timestamp() {
-            tracing::error!("Invalid proposal: l1_finalized timestamp mismatch");
+
+        // Validate `l1_head`.
+        if proposed_header.l1_head() < parent_leaf.block_header().l1_head() {
+            tracing::error!("Invalid proposal: l1_head not increasing");
+            return Err(BlockError::InvalidBlockHeader);
+        }
+
+        let l1_head = instance
+            .l1_client
+            .wait_for_finalized_block(proposed_header.l1_head())
```

Comment:
We don't want to wait for the l1 head to be finalized, we just want to wait for it to exist (ie for the L1 block height to be >= l1_head). Might need a new l1_client method for this.

Reply:
064eda5e3dadd3bdf9a1f71e2bf9b313b2f16914